fhir-model/src/main/java/org/linuxforhealth/fhir/model/string/util/strategy/MaxBytesStringSizeControlStrategy.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * (C) Copyright IBM Corp. 2022
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+package org.linuxforhealth.fhir.model.string.util.strategy;
+
+import java.nio.ByteBuffer;
+import java.nio.CharBuffer;
+import java.nio.charset.Charset;
+import java.nio.charset.CharsetDecoder;
+import java.nio.charset.CodingErrorAction;
+import java.util.Objects;
+import java.util.logging.Logger;
+
+import org.linuxforhealth.fhir.model.string.util.StringSizeControlStrategyFactory.Strategy;
+
+/**
+ * Truncate the input String value to fit the input maxBytes(maximum bytes) size. 
+ */
+public class MaxBytesStringSizeControlStrategy implements StringSizeControlStrategy {
+
+    private static final Logger LOG = Logger.getLogger(MaxBytesStringSizeControlStrategy.class.getName());
+    
+    @Override
+    public String truncateString(String value, int maximumBytes) {
+        
+        LOG.fine(() -> "truncate input string " + value + ", to " + maximumBytes);
+        Objects.requireNonNull(value);
+        Charset charset = Charset.forName("UTF-8");
+        CharsetDecoder charsetDecoder = charset.newDecoder();
+        byte[] byteValue = value.getBytes(charset);
+        if (byteValue.length <= maximumBytes) {
+            return value;
+        }
+        // Ensure truncation by having byteBuffer = maximumBytes
+        ByteBuffer byteBuffer = ByteBuffer.wrap(byteValue, 0, maximumBytes);
+        CharBuffer charBuffer = CharBuffer.allocate(maximumBytes);
```

Comment:
The char buffer length will be at most the same number of characters as bytes (if all characters single-byte). Might be worth adding a comment just to state that you are intentionally using maximumBytes in the context of a character buffer length.

Reply:
This has been fixed as suggested.